### ISSUE TITLE
Please discuss here!

### ISSUE THREAD

**Issue by DanielGibson** [OpenTechEngine/Discussions#1]
To keep the OpenTechBFG issue tracker overseeable, feel free use the issue tracker of this pseudo-project as a discussion forum

**Comment by kortemik** [comment]
Awesome, now we can have can have OpenTechBFG repo just for code issues and discuss the rest here. I always limited the answers because I didn't want to flood the issue tracker but now there is no need for that.

**Comment by BielBdeLuna** [comment]
great, I always found strange that Github doesn't provide with an infrastructure for discussing the projects.

**Comment by ghost** [comment]
Thank you for this, now I can stop flooding issue pages.

**Comment by kortemik** [comment]
http://sourceforge.net/projects/opentechengine/files/snapshot/Windows/ latest builds are now here. 

VisualStudio builds require debug libraries available and that won't be the case unless one has VS(2012?) installed. So only the MinGW ones work w/o it. It can be arranged so that the Release versions are built too but atm it's no-no.

Linux builds atm miss the shared objects for cegui but that can be arranged. I don't see anything gained however from binaries on linux.

**Comment by BielBdeLuna** [comment]
at the moment binaries for Linux make no sense, only when there is a game done with the engine making binaries will make some sort of sense.

**Comment by ghost** [comment]
I've upgraded to VS 2015 community edition, it has a bit more support for the open source community. I ran into issues with VS express 2013 on Win 10 due to some missing linker with kernel32.

**Comment by ghost** [comment]
VS 2015 community edition also has an option for git integration.

It also includes Android development now.

**Comment by kortemik** [comment]
@BielBdeLuna yes, they are only for CI purposes, aka we see the code compiles fine.

The compiling happens here:
https://travis-ci.org/OpenTechEngine/OpenTechBFG (Click on the build job #xxx.1 etc)

And for Visual Studio here:
https://ci.appveyor.com/project/kortemik/opentechbfg

@Yetta1 It's possible to use VS2015 on CI too but would that bring any benefit for us? It's just used for compiling after all. The end binaries should still be the same..

Also we have a static code analysis tool called Coverity in use as you might see on the readme. It scans the code for common issues and creates a report. In case you guys know someone who knows c++ a bit and would like to learn more, resolving the coverity scan resoults would be a soft landing.

I documented these here a while ago: https://github.com/OpenTechEngine/OpenTechBFG/wiki/Workflow-Automation

**Comment by ghost** [comment]
@kortemik I had issues installing VS 2015 Community, so I just fired up Code Blocks with MinGW and opened the .sln from my VS build of OTE and linked the include and lib folders needed. Worked without any issues. I seem to prefer Code Blocks more, but I stick to VS for some reason, which is stupid considering I also use Linux on my other machine with C::B.

**Comment by ghost** [comment]
Heads up.
Intel has a promotional offer on it's Game development tools for the Professional version ending 31 August, it is cross-platform and has some nice tools included such as OpenCL Builder, DirectX/OpenGL analyzers, context sensing and more. 

https://software.intel.com/intel-inde

Contains a version comparison.
https://software.intel.com/en-us/intel-inde/try-buy

**Comment by kortemik** [comment]
I don't think it surprises anyone that I use eclipse. Before that Emacs was my choice but on large scale project like this, emacs plugins I used for code indexing got quite heavy to run. Let's see about that Intel wonder.

**Comment by ghost** [comment]
I'm downloading the Intel software now, 2.1gb installer. I didn't see a linux download, only for OS X and Windows. Mainly grabbing it for the Graphics analyzers and the OpenCL Code builder might come in handy once I understand that. I use to use Eclipse, it's a damn nice IDE, however I ran into some issues with the latest CPP version which is why I went over to C::B.

**Comment by kortemik** [comment]
What kind of problems? I have had none this far, but then again, I use command line tools for anything else than writing/lookups.

**Comment by ghost** [comment]
I can't remember what exactly it was but had issues linking libraries, spent hours looking for an answer on forums and found nothing that worked, eventually I found a post that suggested using C::B and everything worked.

**Comment by ghost** [comment]
@kortemik Intel INDE has direct support for Eclipse, I just read that it includes the Intel Compiler, which i read the other day is quite good. I'll have to install Eclipse again as it seems that the installer for INDE won't recognise VS 2015 community.

**Comment by ghost** [comment]
Vulkan tutorial Github project, still early stages, contains the Mantle API reference and other documentation. Watch the video on the site showcasing the performance of Vulkan on Dota 2 with added creeps in the lane fight using Intel HD graphics.

http://vulkan-tutorial.com/
https://github.com/Overv/VulkanTutorial

**Comment by ghost** [comment]
Is it ok if I point Microsoft to the project to help me with my build issues on VS 2015?

**Comment by kortemik** [comment]
Yes of course it is. We are open source :)

**Comment by BielBdeLuna** [comment]
@Yetta1 do you think they will help you with anything?

**Comment by ghost** [comment]
They are helping me find the bug in VS 2015 on msdn.

**Comment by kortemik** [comment]
I think it's namespace related at least I found multiple bugs on VS2013 but did make some changes so it built and stopped caring..

MSVC needed some strange forward declarations which gcc, clang and mingw were very fine without:
https://github.com/OpenTechEngine/OpenTechBFG/commit/60ff0dff30e37722527451775241ffb0f22ef4e7

**Comment by ghost** [comment]
@kortemik I'm not really sure what namespaces are or how to fix it :/.

Here are some of the errors from the output:

```
15>C:\Users\Tiaan\Documents\GitHub\OpenTechBFG\neo\renderer\Interaction.cpp : fatal error C1001: An internal error has occurred in the compiler.
15>  (compiler file 'f:\dd\vctools\compiler\utc\src\p2\main.c', line 246)
15>   To work around this problem, try simplifying or changing the program near the locations listed above.
15>  Please choose the Technical Support command on the Visual C++
15>   Help menu, or open the Technical Support help file for more information
```

and

```
15>cl : Command line error D8040: error creating or communicating with child process
16>------ Build started: Project: ALL_BUILD, Configuration: Release x64 ------
========== Build: 14 succeeded, 2 failed, 0 up-to-date, 0 skipped ==========
```

I've noticed that VS 2015 did not install some dynamic linked libraries needed for the debug binary you uploaded to the snapshots, msvcr120d.dll and msvcp120d.dll. I downloaded the dll's seperately, however the engine crashes when opened. Only release builds have worked on my system. I have the VC++ redistributables installed, but it also does not include the debug dlls.

I generated the solution with this code in a batch, which has worked.

```
cd ..
del /s /q build
mkdir build
cd build
cmake -G "Visual Studio 14 Win64" -DCMAKE_INSTALL_PREFIX=../bin/win10-64 -DOPENAL=OFF -DBUNDLED_FREETYPE=ON ../
pause
```
